components/centraldashboard/app/api.ts

```diff
@@ -1,23 +1,29 @@
-import express from 'express';
+import express, {Request, Response} from 'express';
 
+import {V1Namespace} from '@kubernetes/client-node';
 import {KubernetesService, PlatformInfo} from './k8s_service';
 import {Interval, MetricsService} from './metrics_service';
-
-interface EnvironmentInfo {
-  namespaces: string[];
-  platform: PlatformInfo;
-  user: string;
-}
+import {DefaultApi, Binding as WorkgroupBinding} from './clients/profile_controller';
 
 const IAP_HEADER = 'X-Goog-Authenticated-User-Email';
 const IAP_PREFIX = 'accounts.google.com:';
 
+interface WorkgroupInfo {
+  namespaces: WorkgroupBinding[];
+  isClusterAdmin: boolean;
+}
+
+interface AuthObject {
+  [IAP_HEADER]: string;
```

Comment:
I do think we should given consideration to @yanniszark 's comment regarding the support for non-Google authentication systems in the future. At the very least, please add a TODO to indicate that it will be addressed in the future.

Reply:
Thanks @prodonjs!
This was also [discussed in the community meeting](https://docs.google.com/document/d/1Wdxt1xedAj7qF_Rjmxy1R0NRdfv7UWs-r2PItewxHpE/edit) and it was agreed that the header/prefix value should be configurable.

Reply:
Thanks, @prodonjs @yanniszark. While Jeremy and the team review the header based approach, I've included a configurable environment flag in af2039aadcd8776318a4d6833d60640abe66d36f.

Reply:
That's great, thanks a lot @avdaredevil! 
Once this is merged and a manifest is made, let's make sure that we make these configurable. kubeflow/manifests#179

Reply:
@avdaredevil can you please rename the env vars to USERID_HEADER and USERID_PREFIX?
IAP is a Google-specific service and doesn't convey the correct meaning.
In addition, this is the name used in other places, like the profile controller.

Reply:
Done! Sorry about that, wasn't aware.